### DISCUSSION TITLE
Including Temporary Files in Quarto and Deleting afterwards

### DISCUSSION THREAD

**Discussion by Jiayou-Chao** [quarto-dev/quarto-cli#10077] [Q&A] [answered]
### Description

I want to generate some intermediate temporary qmd files but delete them afterwards to make my working directory folder clean.
For example, 
````qmd

```{r}
generate_qmd(path = "temporary.qmd")
```

{{< include temporary.qmd >}}

```{r}
file.remove("temporary.qmd")
```
````
However, there will be an error message: "could not find file temporary.qmd." To make things even worse, I have to create a `temporary.qmd` the first time in order for it to run.

Is there a workaround to this? Or should it be a feature request?

**Comment by mcanouil** [comment, marked as answer]
Use project scripts: https://quarto.org/docs/projects/scripts.html#pre-and-post-render

**Reply by Jiayou-Chao** [reply]
The quarto project solution seems to work for me. Thank you.

**Comment by cscheid** [comment]
You're confusing execution orders of different parts of the document. `{{< include >}}` happens for the entire document  _before_ knitr executes the engine cells.
